src/main/java/com/redislabs/redistimeseries/Range.java

```diff
@@ -43,8 +43,10 @@ public String getKey() {
   private static Map<String, String> getLabelsStringStringMap(List<?> resLabels) {
     Map<String, String> rangeLabels = new HashMap<>(resLabels.size());
     for (Object resLabel : resLabels) {
-      List<byte[]> label = (List<byte[]>) resLabel;
-      rangeLabels.put(SafeEncoder.encode(label.get(0)), SafeEncoder.encode(label.get(1)));
+      List<byte[]> pair = (List<byte[]>) resLabel;
+      if (pair.get(0) != null && pair.get(1) != null) {
```

Comment:
1. What if `pair.get(0) != null && pair.get(1) == null`
2. I think we should avoid calling pair.get(0) && pair.get(1) twice for performance

Reply:
1. That's my question too (to RedisTimeSeries developers).
    - In Java Map, setting null value is meaningless (unless you expect to overwrite). As we expect unique keys, I avoided calling a unnecessary Map.put(). This also avoids a unnecessary SafeEncoder.encode(pair.get(0)).
2. ...
    - The performance cost of current code is negligible because underneath is an ArrayList.
    - The perfomance gain of your suggestion is arguable because two new variables would be required (to hold the references) which would later go to gc.

Reply:
> 1. That's my question too (to RedisTimeSeries developers). 

It shouldn't happen, if it happens it's a bug and should throw a parsing error

Reply:
@gkorland I talked with @danni-m, that's the feature.